miniapps/nurbs/nurbs_curveint.cpp

```diff
@@ -0,0 +1,238 @@
+   {
+      patch(1,0,2) = sqrt(2)/2;
+   }
+   patch.KnotInsert(0, *kv);
+
+   if (ifbspline)
+   {
+      // We locate the controlpoints at the location of the
+      // maxima of the knotvectors. This works very well
+      // for patches with the weight being one.
+      kv->FindMaxima(i_args,xi_args, u_args);
+
+      for (int i = 0; i < ncp; i++)
+      {
+         (*xy[0])[i] = u_args[i]*l;
+         (*xy[1])[i] = a * sin((*xy[0])[i]/l*2*3.1415);
+      }
+
+      kv->FindInterpolant(xy);
+
+      // Apply interpolation to patch
+      for (int i = 0; i < ncp; i++)
+      {
+         patch(i,0,0) = (*xy[0])[i]*patch(i,0,2)-0.5*l;
+         patch(i,0,1) = (*xy[1])[i]*patch(i,0,2)-0.5*l;
+      }
+   }
+
+   else
+   {
+      // We use a uniform spacing of the controlpoints. This
+      // works better for patches with not all weights being one
+      // than using the locations of the maxima.
+      for (int i = 0; i < ncp; i++)
+      {
+         (*xy[0])[i] = -0.5+(double) i * l/(ncp-1);
+         (*xy[1])[i] = -a * sin((*xy[0])[i]/l*2*3.1415)-0.5*l;
+      }
+
+      kv->FindInterpolant(xy);
+
+      // Apply interpolation to patch
+      for (int i = 0; i < ncp; i++)
+      {
+         patch(i,0,0) = (*xy[0])[i]*patch(i,0,2);
+         patch(i,0,1) = (*xy[1])[i]*patch(i,0,2);
+      }
+   }
+
+   // Refinement in curve interpolation direction
+   patch.DegreeElevate(1, order-kv_o1->GetOrder());
+   patch.KnotInsert(1, *kv);
+
+   // 3. Open and write mesh output file
+   string mesh_file;
+   mesh_file.append("sin-fit");
+   mesh_file.append(".mesh");
+   ofstream output(mesh_file.c_str());
+
+   output<<"MFEM NURBS mesh v1.0"<<endl;
+   output<< endl << "# Square nurbs mesh with a sine fitted at its bottom edge" <<
+         endl << endl;
+   output<< "dimension"<<endl;
+   output<< 2 <<endl;
+   output<< endl;
+
+   output<<"elements"<<endl;
+   output<<"1"<<endl;
+   output<<"1 3 0 1 2 3"<<endl;
+   output<<endl;
+
+   output<<"boundary"<<endl;
+   output<<"0"<<endl;
+   output<<endl;
+
+   output << "edges" <<endl;
+   output << "4" <<endl;
+   output << "0 0 1"<<endl;
+   output << "0 3 2"<<endl;
+   output << "1 0 3"<<endl;
+   output << "1 1 2"<<endl;
+   output<<endl;
+
+   output << "vertices" << endl;
+   output << 4 << endl;
+
+   output<<"patches"<<endl;
+   output<<endl;
+
+   output << "# Patch 1 " << endl;
+   patch.Print(output);
+   output.close();
+
+   // Print mesh info to screen
+   cout << "=========================================================="<< endl;
+   cout << " Attempting to read mesh: " <<mesh_file.c_str()<< endl ;
+   cout << "=========================================================="<< endl;
+   Mesh *mesh = new Mesh(mesh_file.c_str(), 1, 1);
+   mesh->PrintInfo();
+
+   // Print mesh to file for visualisation
+   VisItDataCollection dc = VisItDataCollection("mesh", mesh);
+   dc.SetPrefixPath("solution");
+   dc.SetCycle(0);
+   dc.SetTime(0.0);
+   dc.Save();
+
+   delete kv_o1, kv;
```

Comment:
The compiler complains about this. Each pointer should have its own `delete` line.

Reply:
Resolved in [e31bfca](https://github.com/mfem/mfem/pull/3232/commits/e31bfca43bf510dfc9e0cd51398ea06efa9707c6)